app/models/talentlms/facade.rb

```diff
@@ -0,0 +1,50 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module Talentlms
+  class Facade
+
+    def initialize
+      @api = Config.first
+    end
+
+    # Login user to TalentLMS
+    #
+    # @@param user [User] the user
+    # @return [String] URL to redirect user to
+    def login(user)
+      login = Login.find_by(user: user)
+      if login.nil?
+        result = create_account(user) if login.nil?
+      else
+        result = @api.post('userlogin', {login: login.login, password: password})
+      end
+      result['login_key']
+    end
+
+    # Create an account in TalentLMS for a user
+    #
+    # @param user [User] the user
+    # @return [String] URL to redirect the user to to login
+    def create_account(user)
+      login = "user_#{user.id}"
+      password = SecureRandom.hex(8)
```

Comment:
Is there a restriction on the length?  If not, let's make this 16 or even 20.

Reply:
Yes, there is a restriction. I determined this experimentally, but I don't remember now... I think it was 16 characters, hence, 8 bytes...